crux-core/src/crux/query_state.clj

```diff
@@ -1,28 +1,58 @@
 (ns ^:no-doc crux.query-state
-  (:import (crux.api QueryState QueryState$QueryStatus QueryState$QueryError)))
+  (:import (crux.api IQueryState IQueryState$IQueryError IQueryState$QueryStatus)
+           (java.io Writer)))
+
+(defn ->query-status [status]
+  (case status
+    :failed IQueryState$QueryStatus/FAILED
+    :completed IQueryState$QueryStatus/COMPLETED
+    :in-progress IQueryState$QueryStatus/IN_PROGRESS))
+
+(defrecord QueryState$QueryError [type message]
```

Comment:
I've not seen `$` in Clojure record names before, didn't even know it would be valid! I'd stick with just `QueryError`, the namespace makes it sufficiently unambiguous. (usages too)

```suggestion
(defrecord QueryError [type message]
```

Reply:
I was going based off how the inner interface was imported. I'll change it to just be QueryError